homeassistant/components/slide_local/cover.py

```diff
@@ -0,0 +1,307 @@
+    """Set up cover(s) for Slide platform."""
+
+    _LOGGER.debug("Initializing Slide cover(s)")
+
+    platform = entity_platform.async_get_current_platform()
+    platform.async_register_entity_service(
+        SERVICE_CALIBRATE,
+        None,
+        "async_calibrate",
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_TOUCHGO,
+        {vol.Required(ATTR_TOUCHGO): cv.boolean},
+        "async_touchgo",
+    )
+
+    _LOGGER.debug(
+        "Trying to setup Slide '%s', config=%s",
+        entry.data[CONF_HOST],
+        str(entry),
+    )
+
+    await entry.runtime_data.api.slide_add(
+        entry.data[CONF_HOST],
+        entry.data[CONF_PASSWORD],
+        entry.data[CONF_API_VERSION],
+    )
+
+    try:
+        slide_info = await entry.runtime_data.api.slide_info(entry.runtime_data.host)
+    except (ClientConnectionError, ClientTimeoutError) as err:
+        # https://developers.home-assistant.io/docs/integration_setup_failures/
+
+        _LOGGER.error(
+            "Unable to get information from Slide '%s': %s",
+            entry.runtime_data.host,
+            str(err),
+        )
+
+    if slide_info is None:
+        _LOGGER.error("Unable to setup Slide '%s'", entry.runtime_data.host)
+    elif slide_info.get("mac") is None:
+        _LOGGER.error(
+            "Unable to setup Slide Local '%s', the MAC is missing in the slide response (%s)",
+            entry.data[CONF_HOST],
+            slide_info,
+        )
+    else:
+        async_add_entities(
+            [
+                SlideCoverLocal(
+                    entry.runtime_data.api,
+                    slide_info,
+                    entry.runtime_data.host,
+                    entry.data[CONF_MAC],
+                    entry.data[CONF_INVERT_POSITION],
+                )
+            ]
+        )
+
+        _LOGGER.debug("Setup Slide '%s' successful", entry.runtime_data.host)
+
+
+class SlideCoverLocal(CoverEntity):
+    """Representation of a Slide Local API cover."""
+
+    _attr_assumed_state = True
+    _attr_device_class = CoverDeviceClass.CURTAIN
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        api: SlideLocalApi,
+        slide_info: dict[str, Any],
+        host: str,
+        mac: str,
+        invert: bool,
+    ) -> None:
+        """Initialize the cover."""
+        self._api = api
+        self._id = host
+        self._invert = invert
+        self._name = slide_info.get("device_name", host)
+        self._slide: dict[str, Any] = {}
+        self._slide["pos"] = None
+        self._slide["state"] = None
+        self._slide["online"] = False
+        self._slide["touchgo"] = False
+        self._unique_id = mac
+
+        self.parsedata(slide_info)
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return the device unique id."""
+        return self._unique_id
+
+    @property
+    def name(self) -> str:
+        """Return the device name."""
+        return self._name
+
```

Comment:
These can be shorthand

Reply:
What do you mean by that?

Reply:
most of the properties have a default to return `self._attr_name` for example, this way you don't need the little intermediate attributes and return them in properties

Reply:
ah, you meant shortened not shorthand...

Reply:
I did mean shorthand, that's how we call them :)

Reply:
Changed it for the name.
What do you think, do we even need the extra attributes?

Reply:
> I did mean shorthand, that's how we call them :)

Ok, again learned something. Thnx.

Reply:
The extra state attributes? I am not sure what the `_id` is, but I think we can remove them, also because touchgo will be its own switch I assume

Reply:
id is the IP. So I will remove it then.